control-plane/aegisops_control_plane/execution_coordinator.py

```
@@ -0,0 +1,1077 @@
+                "shuffle receipt does not match approved delegation binding"
+            )
+
+    def _finalized_execution_provenance(
+        self,
+        *,
+        execution: ActionExecutionRecord,
+        receipt: object,
+        execution_surface_id: str,
+    ) -> dict[str, object]:
+        provenance = dict(execution.provenance)
+        provenance["adapter"] = getattr(receipt, "adapter")
```

Ruff (0.15.9)

[warning] Do not call `getattr` with a constant attribute value. It is not any safer than normal property access.

Replace `getattr` with attribute access

(B009)